### PR TITLE
UIPBEX-53 Implemented TitleManager component

### PR DESCRIPTION
**Pull request by chris-hellen**
## Purpose
Added descriptive HTML title to 'Transfer criteria' section of Users settings so that users have easier time navigating multiple pages.

https://folio-org.atlassian.net/jira/software/c/projects/UIPBEX/issues/UIPBEX-53?jql=project%20IN%20%28UIPBEX%2C%20UICAL%29%20ORDER%20BY%20created%20DESC%2C%20status%20DESC

## Approach
Used TitleManager component to implement the descriptive title.

### CODE REVIEW

File: src/BursarExportPlugin.tsx [hunk 1]
```diff
@@ -71,6 +71,8 @@ export default function BursarExportPlugin() {
   }
 
   return (
+    <>
+    <TitleManager page={useIntl().formatMessage({ id: "Users settings - Transfer criteria" })}></TitleManager>
```
Comment (danetsao) [inline src/BursarExportPlugin.tsx:75]: To use `useIntl` in this case, you want to have `const intl = useIntl();` at top of file, below `const stripes = useStripes();`. Then you can call `useIntl.formatMessage`.

For `formatMessage`, this takes an id of a string, so you can name it something like `ui-plugin-bursar-export.meta.settingTitle` or smth and then put the string you have right now in the `translations/en.json` file.

And then the `TitleManager` can just wrap the whole thing your returning and you don't need the `<> </>`.

**Review by danetsao — CHANGES_REQUESTED** [review]
And then looks like some linting issues and one failing test when I tested it locally, so make sure to run tests and lints.

Also, last thing, do you have your formatting set up? When I pulled local, eslint was showing whitespacing errors, may also happen on lint.

File: src/BursarExportPlugin.tsx [hunk 2]
```diff
@@ -71,17 +73,19 @@ export default function BursarExportPlugin() {
...
-      id="pane-batch-group-configuration"
-      paneTitle={<FormattedMessage id="ui-plugin-bursar-export.bursarExports.paneTitle" />}
-    >
-      <ConfigurationForm
-        initialValues={initialValues}
-        onSubmit={submitCallback}
-        formApiRef={formApiRef}
-      />
-    </Pane>
+    <TitleManager page={intl.formatMessage({ id: 'ui-plugin-bursar-export.meta.settingsTitle' })}>
```
Comment (danetsao) [inline src/BursarExportPlugin.tsx:76]: Implementation looks good here. Although I think you should pass `stripes={stripes}` in as is done in the [ui-cal implementation](https://github.com/folio-org/ui-calendar/pull/502).

**Review by danetsao — CHANGES_REQUESTED** [review]
Implementation looks good, but is there a reason you made the `TitleManager` component in this repo? I assume it's bc the `BursarExportPlugin` was not rendering properly, but I investigated and this is bc we have to mock the `TitleManager` component in the jest test. You can do that with something like this I believe.

Screenshot 2024-05-15 at 3 34 14 PM

**Comment by chris-hellen** [comment]
Yes, I was having an issue with the bursar plugin rendering. Kept saying that the imports weren't correct, and figured it was because the original "TitleManager" didn't have a specific type since it was JS. I'll try mocking TitleManager with the original implementation. Thanks.

**Review by danetsao — APPROVED** [review]
Looks good to me! I requested review from @ncovercash as well.

File: translations/ui-plugin-bursar-export/en.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "meta.title": "Transfer configuration",
+  "meta.settingsTitle": "Users settings - Transfer criteria",
```
Comment (ncovercash) [inline translations/ui-plugin-bursar-export/en.json:3]: Meta is special for the settings pane title only - use something else here (title.settings or something)

Doesn't matter as much for this, but still probably would be good

**Review by ncovercash — APPROVED** [review]

**Pull request merged**